Mathlib/NumberTheory/Cyclotomic/Rat.lean

```diff
@@ -422,6 +422,80 @@ lemma prime_norm_toInteger_sub_one_of_prime_ne_two' [hcycl : IsCyclotomicExtensi
   replace hζ : IsPrimitiveRoot ζ (p ^ (0 + 1)) := by simpa using hζ
   exact hζ.prime_norm_toInteger_sub_one_of_prime_ne_two hodd
 
+/-- In a `p ^ (k + 1)`-th cyclotomic extension of `ℚ `, we have that `ζ` is not congruent to an
+  integer modulo `p` if `p ^ (k  + 1) ≠ 2`. -/
+theorem not_exists_int_prime_dvd_sub_of_prime_pow_ne_two
+    [hcycl : IsCyclotomicExtension {p ^ (k + 1)} ℚ K]
+    (hζ : IsPrimitiveRoot ζ ↑(p ^ (k + 1))) (htwo : p ^ (k + 1) ≠ 2) :
+    ¬(∃ n : ℤ, (p : 𝓞 K) ∣ (hζ.toInteger - n : 𝓞 K)) := by
+  intro ⟨n, x, h⟩
+  let pB := hζ.integralPowerBasis
+  have hdim : pB.dim = ↑p ^ k * (↑p - 1) := by
+    simp [integralPowerBasis_dim, pB, Nat.totient_prime_pow hp.1 (Nat.zero_lt_succ k)]
+  replace hdim : 1 < pB.dim := by
+    by_cases hk : k = 0
+    · rw [hdim, hk, pow_zero, one_mul]
+      contrapose htwo
+      simp only [hk, zero_add, pow_one, ne_eq, Decidable.not_not]
+      rcases lt_or_eq_of_le (Nat.Prime.two_le hp.1) with (h | h)
+      · omega
+      · exact_mod_cast h.symm
+    · rw [hdim]
+      exact one_lt_mul_of_lt_of_le (one_lt_pow hp.1.one_lt hk)
+        (have := Nat.Prime.two_le hp.1; by omega)
+  rw [sub_eq_iff_eq_add] at h
+  replace h := pB.basis.ext_elem_iff.1 h ⟨1, hdim⟩
```

Comment:
Until here nothing happened, but this point contains "an idea", I would say. Can you add a bit of doc along the proof?

Reply:
Done, thanks!